lib/review-apps.unit.test.js

```diff
@@ -217,183 +406,111 @@ describe('review-apps', () => {
 						}
 					}
 				]);
+
 			const build = await getAppBuildWithCommit({ appId, commit });
 			expect(build).toBeFalsy();
+
 		});
 
 		it('returns undefined if there is no builds', async () => {
+
 			const appId = 'app123';
 			const commit = 'ababa';
+
 			nockScope.get('/apps/app123/builds')
 				.reply(200, []);
+
 			const build = await getAppBuildWithCommit({ appId, commit });
 			expect(build).toBeFalsy();
+
 		});
+
 	});
 
-	describe('waitForReviewAppBuild', () => {
-		const { waitForReviewAppBuild } = reviewApps;
+	describe('repeatedCheckForBuildSuccessStatus', () => {
+
+		const { repeatedCheckForBuildSuccessStatus } = reviewApps;
+
+		it('returns without throwing an error if the app build is returned with a status of \'succeeded\'', async () => {
 
-		it('returns appId of succeeded review app build', async () => {
 			const commit = 'a00000000000';
 			const appId = 'app123';
 			const reviewAppBuild = {
-				app: {
-					id: appId
-				},
 				source_blob: {
 					version: commit
 				},
 				status: 'succeeded'
 			};
-			nockScope.get('/apps/app123/builds')
-				.reply(200, [ reviewAppBuild ]);
-
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit
-			})(appId);
 
-			expect(waitedReviewAppBuild).toEqual(appId);
-		});
+			nockScope.get('/apps/app123/builds')
+				.reply(200, [reviewAppBuild]);
 
-		it('waits for succeeded review app build from no review apps', async () => {
-			const commit = 'a00000000000';
-			const appId = 'app123';
-			const succeededReviewAppBuild = {
-				app: {
-					id: appId
-				},
-				source_blob: {
-					version: commit
-				},
-				status: 'succeeded'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(200, [])
-				.get('/apps/app123/builds')
-				.reply(200, [ succeededReviewAppBuild ]);
+			try {
+				await repeatedCheckForBuildSuccessStatus({ commit, appId });
+			} catch (error) {
+				expect(true).toBeFalsy();
+			}
 
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit,
-				minTimeout: 0
-			})(appId);
+			expect(true).toBeTruthy();
```

Comment:
This will always pass

Reply:
I added console logs just before `expect(true).toBeFalsy();` and `expect(true).toBeTruthy();` respectively.

If an error is thrown by `repeatedCheckForBuildSuccessStatus()` then it is caught and the expectation that always fails will run and so cause the test to fail. In this scenario, `expect(true).toBeTruthy();` is not reached, and even if it were, the expectation that always fails will also have been encountered and cause the test to fail overall.

```js
try {
	await repeatedCheckForBuildSuccessStatus({ commit, appId });
} catch (error) {
	expect(true).toBeFalsy();
}

expect(true).toBeTruthy();
```

It's not ideal as it's not particularly explicit as to the intention of the test, but I could not get this working with `expect(() => ...error...).not.toThrow(error)` when accommodating an async function (re. https://stackoverflow.com/questions/49603338/how-to-test-an-exception-was-not-thrown-with-jest#answer-49603347). I think this would be a preferable solution if we can get it working.